src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -89,15 +93,26 @@ static void RegisterChannels(FeatureConfigurationContext context, IManageReceive
             {
                 channelReceiverFactory = context.Settings.Get<Func<string, IChannelReceiver>>("GatewayChannelReceiverFactory");
                 channelSenderFactory = context.Settings.Get<Func<string, IChannelSender>>("GatewayChannelSenderFactory");
+                return;
             }
-            else
+
+            CheckForNonWildcardDefaultChannel(channelManager);
+
+            channelReceiverFactory = s => new ChannelReceiverFactory(typeof(HttpChannelReceiver)).GetReceiver(s);
+            channelSenderFactory = s => new ChannelSenderFactory(typeof(HttpChannelSender)).GetSender(s);
+
+            var installerSettings = context.Settings.Get<InstallerSettings>();
+            installerSettings.ChannelManager = channelManager;
+            installerSettings.Enabled = true;
+        }
+
+        static void CheckForNonWildcardDefaultChannel(IManageReceiveChannels channelManager)
+        {
+            var defaultChannel = channelManager.GetDefaultChannel();
+            if (defaultChannel.Address.Contains("*") || defaultChannel.Address.Contains("+"))
```

Comment:
Don't we have to check if this is a http channel?

Reply:
Nah, it is implicit. This method is only called if you are running with the default http channel manager.